python/pyspark/pandas/namespace.py

```diff
@@ -2814,9 +2824,18 @@ def to_numeric(arg):
     1.0
     """
     if isinstance(arg, Series):
-        return arg._with_new_scol(arg.spark.column.cast("float"))
+        if errors == "coerce":
+            return arg._with_new_scol(arg.spark.column.cast("int"))
+        elif errors == "ignore":
+            scol = arg.spark.column
+            casted_scol = scol.cast("int")
+            return arg._with_new_scol(F.when(casted_scol.isNull(), scol).otherwise(casted_scol))
```

Comment:
If `scol` is a string column, the output type can be a string column:

```python
>>> from pyspark.sql import functions as F
>>> scol = F.col("a")
>>> casted_scol = scol.cast("int")
>>> df = sql("SELECT 'a' as a")
>>> df.select(F.when(casted_scol.isNull(), scol).otherwise(casted_scol)).printSchema()
root
 |-- CASE WHEN (CAST(a AS INT) IS NULL) THEN a ELSE CAST(a AS INT) END: string (nullable = true)
```

is this correct type?

Reply:
Yeah I think so.

If the `errors` is "ignore", then the original data is returned as is:

```python
>>> pd.to_numeric(pd.Series(["1", "2", "hello"]), errors="ignore")
0        1
1        2
2    hello
dtype: object
```

Reply:
when errors is `ignore`, it should return the input as is per the documentation. The problem is the type coercion will happen via Spark. For example, it throws an exception if you call `pd.to_numeric` with `datetime.datetime`s whereas pandas returns the input as is.

Reply:
can you try this:

```python
pd.to_numeric(pd.Series([datetime.datetime(1970, 1, 2)]), errors="ignore").to_list()
```

Reply:
Actually the case @itholic raised is a bit tricky.

pandas can return numeric type if there is no error.

```py
>>> pd.to_numeric(pd.Series(["1", "2", "3"]), errors="ignore")
0    1
1    2
2    3
dtype: int64
```

whereas the current implementation always returns `StringType`:

```py
>>> ps.to_numeric(ps.Series(["1", "2", "3"]), errors="ignore")
0    1
1    2
2    3
dtype: object
```

As Spark can't change the data type depending on whether there is an error or not, we have to check it by ourselves beforehand. (or just we don't support this?)

Reply:
I think we don't support this for `Series` for now, otherwise we should check all types in the `Series` and nullability.

Reply:
Yeah, we can don't implement it for now.